exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/efgs/OperationDao.java

```diff
@@ -1,161 +1,127 @@
+        return jdbcTemplate.update(sql, params) == 1;
     }
 
-    public boolean finishOperation(long operationId, int keysCountTotal, int keysCount201, int keysCount409, int keysCount500) {
-        String sql = "update en.efgs_operation set state = cast(:new_state as en.state_t), " +
+    public boolean finishOperation(FederationOutboundOperation operation, int keysCountTotal, int keysCount201, int keysCount409, int keysCount500) {
+        String sql = "update en.efgs_operation set state = cast(:new_state as en.state_t), batch_tag = :batch_tag, " +
                 "keys_count_total = :keys_count_total, keys_count_201 = :keys_count_201, " +
-                "keys_count_409 = :keys_count_409, keys_count_500 = :keys_count_500, " +
-                "run_count = run_count + 1 " +
+                "keys_count_409 = :keys_count_409, keys_count_500 = :keys_count_500, updated_at = :updated_at " +
                 "where id = :id";
         return jdbcTemplate.update(sql, Map.of(
                 "new_state", EfgsOperationState.FINISHED.name(),
-                "id", operationId,
+                "id", operation.operationId,
+                "batch_tag", operation.batchTag,
                 "keys_count_total", keysCountTotal,
                 "keys_count_201", keysCount201,
                 "keys_count_409", keysCount409,
-                "keys_count_500", keysCount500
+                "keys_count_500", keysCount500,
+                "updated_at", new Timestamp(Instant.now().toEpochMilli())
         )) == 1;
     }
 
-    public void markErrorOperation(long operationId) {
-        String sql = "update en.efgs_operation set state = cast(:error_state as en.state_t), updated_at = :updated_at, " +
-                "run_count = run_count + 1 " +
-                "where id = :id";
-        jdbcTemplate.update(sql, Map.of(
-                "error_state", EfgsOperationState.ERROR.name(),
-                "updated_at", new Timestamp(Instant.now().toEpochMilli()),
-                "id", operationId
-        ));
+    public void markErrorOperation(long operationId, Optional<String> batchTag) {
+        markErrorOperation(operationId, batchTag, new Timestamp(Instant.now().toEpochMilli()));
     }
 
-    public void setStalledToError() {
-        String sql = "update en.efgs_operation set state = cast(:error_state as en.state_t), updated_at = :updated_at " +
-                "where state = cast(:started_state as en.state_t) and (now() > (updated_at + interval '10 minute')) ";
-        jdbcTemplate.update(sql, Map.of(
-                "error_state", EfgsOperationState.ERROR.name(),
-                "started_state", EfgsOperationState.STARTED.name(),
-                "updated_at", new Timestamp(Instant.now().toEpochMilli())
-        ));
+    public void markErrorOperation(long operationId, Optional<String> batchTag, Timestamp timestamp) {
+        String sql = "update en.efgs_operation set state = cast(:error_state as en.state_t), updated_at = :updated_at, " +
```

Comment:
For clarity, you could break these sql-lines so each assignment (x = :x) is on a separate line when there's more than 1

Reply:
Yep, I have done it now for that specific line. Check others later.